File: hw/fuentes/U3/HW17Maze/MazeAppPY/ec/edu/espe/Maze/model/Door.py
from abc import ABC, abstractmethod

class Door(ABC):
    def __init__(self, from_room, to_room):
        self.from_room = from_room
        self.to_room = to_room

    @abstractmethod
    def get_type(self):
        pass

    @abstractmethod
    def is_exterior(self):
        pass

    def get_from(self):
        return self.from_room

    def get_to(self):
        return self.to_room

File: hw/fuentes/U3/HW17Maze/NewMazeGenerator/src/main/java/ec/edu/espe/mazegenerator/model/Door.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.mazegenerator.model;

/**
 *
 * @author jorge
 */
public abstract class Door {
    protected Room from;
    protected Room to;

    public Door(Room from, Room to) {
        this.from = from;
        this.to = to;
    }

    public abstract String getType();
    public abstract boolean isExterior();

    public Room getFrom(){ 
        return from; 
    }
    public Room getTo(){ 
        return to; 
    }
    
}
